File: RokidSDKDemoJava/app/src/main/java/com/rokid/mobile/sdk/demo/java/adapter/item/BleItem.java
package com.rokid.mobile.sdk.demo.java.adapter.item;

import android.text.TextUtils;
import android.view.View;
import android.widget.ProgressBar;
import android.widget.TextView;

import com.rokid.mobile.lib.entity.bean.bluetooth.BTDeviceBean;
import com.rokid.mobile.sdk.demo.java.R;
import com.rokid.mobile.ui.recyclerview.item.BaseItem;
import com.rokid.mobile.ui.recyclerview.item.BaseViewHolder;

import butterknife.BindView;
import butterknife.ButterKnife;

/**
 * Created by tt on 2018/2/25.
 */

public class BleItem extends BaseItem<BTDeviceBean> {

    @BindView(R.id.item_bind_ble_name_txt)
    TextView bleNameTxt;

    @BindView(R.id.item_bind_ble_bind_txt)
    TextView bindTxt;

    @BindView(R.id.item_bind_ble_progress)
    ProgressBar progressBar;

    public BleItem(BTDeviceBean data) {
        super(data);
    }

    @Override
    public int getViewType() {
        return 0;
    }

    @Override
    public int getLayoutId(int viewType) {
        return R.layout.item_bind_ble;
    }

    @Override
    public void onReleaseViews(BaseViewHolder holder, int sectionKey, int sectionViewPosition) {
        ButterKnife.bind(this, holder.getItemView());
        bleNameTxt.setText("");
        bindTxt.setVisibility(View.GONE);
        progressBar.setVisibility(View.GONE);
    }

    @Override
    public void onSetViewsData(BaseViewHolder holder, int sectionKey, int sectionViewPosition) {
        if (!TextUtils.isEmpty(getData().getName())) {
            bleNameTxt.setText(getData().getName());
        }
    }

    public void showProgressBar() {
        progressBar.setVisibility(View.VISIBLE);
    }

    public void hideProgressBar() {
        progressBar.setVisibility(View.GONE);
    }

    public void showBindTxt() {
        bindTxt.setVisibility(View.VISIBLE);
    }

    public void hideBindTxt() {
        bindTxt.setVisibility(View.GONE);
    }
}

File: RokidSDKDemokotlin/app/src/main/kotlin/com/rokid/mobile/sdk/demo/adapter/item/BleItem.kt
package com.rokid.mobile.sdk.demo.adapter.item

import android.view.View
import com.rokid.lib_appbase.recyclerview.item.BaseItem
import com.rokid.lib_appbase.recyclerview.item.BaseViewHolder
import com.rokid.mobile.lib.entity.bean.bluetooth.BTDeviceBean
import com.rokid.mobile.sdk.demo.R
import kotlinx.android.synthetic.main.item_ble.view.*

/**
 * Created by wangshuwen on 2017/12/8.
 */
class BleItem(data: BTDeviceBean) : BaseItem<BTDeviceBean>(data) {

    val BLE_ITEM_TYPE = 0

    override fun getViewType(): Int = BLE_ITEM_TYPE

    override fun getLayoutId(viewType: Int): Int = R.layout.item_ble

    override fun onReleaseViews(holder: BaseViewHolder?, sectionKey: Int, sectionViewPosition: Int) {
        holder!!.itemView.apply {
            ble_item_btName_txt.text = ""
            ble_item_progress.visibility = View.GONE
        }
    }

    override fun onSetViewsData(holder: BaseViewHolder?, sectionKey: Int, sectionViewPosition: Int) {
        holder!!.itemView.apply {
            ble_item_btName_txt.text = data.name
        }
    }


    fun showProgress() {
        if (null == holder.itemView.ble_item_progress) {
            return
        }
        holder.itemView.ble_item_progress.visibility=View.VISIBLE
    }

    fun hideProgress() {
        if (null == holder.itemView.ble_item_progress) {
            return
        }
        holder.itemView.ble_item_progress.visibility=View.GONE
    }

}